File: woodroid/src/main/java/me/gilo/woodroid/models/filters/OrderFilter.java
package me.gilo.woodroid.models.filters;

public class OrderFilter extends ListFilter{

    private int[] parent;
    private int[] parent_exclude;

    String status;

    int customer;
    int product;
    int dp;

    public int[] getParent() {
        return parent;
    }

    public void setParent(int[] parent) {
        this.parent = parent;
        addFilter("parent", parent);
    }

    public int[] getParent_exclude() {
        return parent_exclude;
    }

    public void setParent_exclude(int[] parent_exclude) {
        this.parent_exclude = parent_exclude;

        addFilter("parent_exclude", parent_exclude);
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;

        addFilter("status", status);
    }

    public int getCustomer() {
        return customer;
    }

    public void setCustomer(int customer) {
        this.customer = customer;

        addFilter("customer", customer);
    }

    public int getProduct() {
        return product;
    }

    public void setProduct(int product) {
        this.product = product;

        addFilter("product", product);
    }

    public int getDp() {
        return dp;
    }

    public void setDp(int dp) {
        this.dp = dp;
        addFilter("dp", dp);

    }
}

File: woodroid/src/main/java/me/gilo/woodroid/models/filters/OrderFilter.kt
package me.gilo.woodroid.models.filters

class OrderFilter : ListFilter() {

    var parent: IntArray? = null
        set(parent) {
            field = parent
            if (parent != null) {
                addFilter("parent", parent)
            }
        }
    var parent_exclude: IntArray? = null
        set(parent_exclude) {
            field = parent_exclude

            if (parent_exclude != null) {
                addFilter("parent_exclude", parent_exclude)
            }
        }

    internal lateinit var status: String

    internal var customer: Int = 0
    internal var product: Int = 0
    internal var dp: Int = 0

    fun getStatus(): String {
        return status
    }

    fun setStatus(status: String) {
        this.status = status

        addFilter("status", status)
    }

    fun getCustomer(): Int {
        return customer
    }

    fun setCustomer(customer: Int) {
        this.customer = customer

        addFilter("customer", customer)
    }

    fun getProduct(): Int {
        return product
    }

    fun setProduct(product: Int) {
        this.product = product

        addFilter("product", product)
    }

    fun getDp(): Int {
        return dp
    }

    fun setDp(dp: Int) {
        this.dp = dp
        addFilter("dp", dp)

    }
}
